pkg/client/client.go

```diff
@@ -988,7 +988,7 @@ func (c *Client) WaitForDaemonSetRollout(ctx context.Context, ds *appsv1.DaemonS
 		}
 
 		var nodeReadyCount int32
-		nodeList, err := c.kclient.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+		nodeList, err := c.kclient.CoreV1().Nodes().List(ctx, metav1.ListOptions{LabelSelector: "kubernetes.io/os=linux"})
```

Comment:
it would be better to use `ds.Spec.Template.Spec.NodeSelector`

Reply:
done